lib/tlscontext.h

```diff
@@ -44,6 +44,16 @@ typedef enum
   TVM_REQUIRED=0x0020,
 } TLSVerifyMode;
 
+typedef enum
+{
+  TSO_NONE,
+  TSO_NOSSLv2=0x0001,
+  TSO_NOSSLv3=0x0002,
+  TSO_NOTLSv1=0x0004,
+  TSO_NOTLSv11=0x0008,
+  TSO_NOTLSv12=0x0010,
+} TLSSslOptions;
+
```

Comment:
I would slightly disagree. Enums are available in a debugger while defines
are not.
On May 20, 2015 4:55 PM, "juhaszviktor" <notifications@github.com> wrote:

> In lib/tlscontext.h
> <https://github.com/balabit/syslog-ng/pull/457#discussion_r30710623>:
>
> > @@ -44,6 +44,16 @@ typedef enum
> >    TVM_REQUIRED=0x0020,
> >  } TLSVerifyMode;
> >
> > +typedef enum
> > +{
> > +  TSO_NONE,
> > +  TSO_NOSSLv2=0x0001,
> > +  TSO_NOSSLv3=0x0002,
> > +  TSO_NOTLSv1=0x0004,
> > +  TSO_NOTLSv11=0x0008,
> > +  TSO_NOTLSv12=0x0010,
> > +} TLSSslOptions;
> > +
>
> You use TlsSslOption as a flag, but the type of it is enumeration.
> Enumerations are usually used for unique values, not for combinations of
> these.
> I mean, that you should "typedef unsigned int TlsSslOptions" and declare
> the TSO_* as define (e.g.: #define TSO_NOSSLv2 1)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/457/files#r30710623>.
>